include/swift/SIL/MemAccessUtils.h

```diff
@@ -1497,35 +1499,54 @@ Result AccessUseDefChainVisitor<Impl, Result>::visit(SILValue sourceAddr) {
 
 namespace swift {
 
-/// Clone all projections and casts on the access use-def chain until either the
-/// specified predicate is true or the access base is reached.
+/// Clone all projections and casts on the access use-def chain until the
+/// checkBase predicate returns a valid base.
```

Comment:
In a subsequent commit, can you describe the function signature of checkBase here and specify what it means to return a "valid" base. Without that I don't know what that means. I would maybe do something like this:

Clone all projections and cases on the access use-def chain until the checkBase predicate returns a valid SILValue as a base.

Reply:
The next line below says "See comments on the cloneUseDefChain() API."